sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDistinctAggregates.scala

```diff
@@ -194,7 +238,11 @@ object RewriteDistinctAggregates extends Rule[LogicalPlan] {
       val regularAggOperatorMap = regularAggExprs.map { e =>
         // Perform the actual aggregation in the initial aggregate.
         val af = patchAggregateFunctionChildren(e.aggregateFunction)(regularAggChildAttrLookup.get)
-        val operator = Alias(e.copy(aggregateFunction = af), e.sql)()
+        val filterOpt = e.filter.map { fe =>
+          val newChildren = fe.children.map(c => regularAggChildAttrLookup.getOrElse(c, c))
+          fe.withNewChildren(newChildren)
```

Comment:
do we still need this change?

Reply:
Yes. We need transform filter expression to attrs.

Reply:
what do you mean? The `Expand` produces attributes and the filter can be evaluated directly.

Reply:
ah I got it after reading `expressionAttributePair`

We change the attributes in the `Expand` output so we need to replace the attributes in filter conditions with new ones.

But the code here is wrong. `.children` doesn't always give us attributes, e.g. `a + b > 1`. We should do
```
e.filter.map(_.transform {
  case a: Attribute => regularAggChildAttrLookup.getOrElse(a, a)
})
```

Reply:
OK. Thanks for your remind and I learned more.